internal/controller/controllers/agent_controller.go

```diff
@@ -505,6 +524,14 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, agent *aiv1beta1.A
 		return err
 	}
 
+	err = r.updateHostIgnition(ctx, c, internalHost, agent)
+	if err != nil {
+		if gorm.IsRecordNotFoundError(err) {
```

Comment:
not sure what's the difference but all other use" if errors.Is(err, gorm.ErrRecordNotFound)"